core/shared/src/main/scala/zio/internal/SingleThreadedRingBuffer.scala

```diff
@@ -28,6 +28,12 @@ private[zio] final class SingleThreadedRingBuffer[A](capacity: Int) {
     increment()
   }
 
+  def putAndGet(value: A): Option[A] = {
```

Comment:
This feels a little bit uncompositional but can address in a follow up.

Reply:
@adamgfraser any suggestions? I can address this in this PR if you like.

Reply:
@luis3m Basically just separate putting and getting into their own operators instead of having one that does both.